daffodil-sapi/src/test/scala/org/apache/daffodil/example/TestScalaAPI.scala

```diff
@@ -1174,4 +1178,100 @@ class TestScalaAPI {
     Daffodil.setLoggingLevel(LogLevel.Info)
   }
 
+  @Test
+  def testScalaAPI24(): Unit = {
+    // Demonstrates error cases of setting external variables
+    val c = Daffodil.compiler()
+
+    val schemaFile = getResource("/test/sapi/mySchemaWithComplexVars1.dfdl.xsd")
+    val pf = c.compileFile(schemaFile)
+    var dp = pf.onPath("/")
+    dp = reserializeDataProcessor(dp)
+
+    // set var without a namespace, ambiguity error because schema contains
+    // two variables with same name but different namespace
+    try {
+      dp = dp.withExternalVariables(Map("var" -> "10"))
+    } catch {
+      case e: ExternalVariableException => {
+        assertTrue(e.getMessage().contains("var"))
```

Comment:
Suggest insert `val msg = e.getMessage()`  just to make code cleaner if this code is copied as a pattern.

Reply:
Will do.